app/models/concerns/rdf/graphable/dereferenceable.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RDF
+  module Graphable
+    # Derference RDF statements if their value matches the URLs in DEREFERENCEABLE_URLS
+    #
+    # Runs during the +graph+ callback.
+    #
+    # @example
+    #   class MyDocument
+    #     include Mongoid::Document
+    #     include RDF::Graphable
+    #     include RDF::Graphable::Dereferenceable
+    #
+    #     field :dcterms_spatial, type: ArrayOf.type(String), default: []
+    #
+    #     dereferences RDF::Vocab::DC.spatial
+    #   end
+    #
+    #   doc = MyDocument.new(dcterms_spatial: 'htttp://www.example.org/rdf_element)
+    #   doc.graph
+    #   doc.rdf_graph.query(predicate: RDF::Vocab::DC.spatial) #=> rdf
+    module Dereferenceable
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def dereferences(*predicates, **options)
+          class_eval do
+            predicates.each do |predicate|
+              callback_proc = proc { dereference_rdf_graph!(predicate) }
+              set_callback :graph, :after, callback_proc, options
+            end
+          end
+        end
+      end
+
+      DEREFERENCEABLE_URLS = %w(http://data.europeana.eu/place/base http://data.europeana.eu/concept/base).freeze
```

Comment:
This reduces the reusability of this module.

I'd be tempted to instead make it an option to `.dereferences`, e.g.
```ruby
  class MyDocument
    include Mongoid::Document
    include RDF::Graphable
    include RDF::Graphable::Dereferenceable

    field :dcterms_spatial, type: ArrayOf.type(String), default: []
    field :dc_type, type: ArrayOf.type(String), default: []

    dereferences RDF::Vocab::DC.spatial, only: %r(\Ahttp://data.europeana.eu/place/base)
    dereferences RDF::Vocab::DC11.type, only: %r(\Ahttp://data.europeana.eu/concept/base)
  end
```